src/main/java/dev/lambdaurora/lambdabettergrass/metadata/LBGGrassState.java

```diff
@@ -78,10 +78,10 @@ public LBGGrassState(@NotNull Identifier id, @NotNull ResourceManager resourceMa
 	 */
 	private @Nullable LBGMetadata loadMetadata(@NotNull ResourceManager resourceManager, @NotNull Identifier metadataId) {
 		var metadataResourceId = new Identifier(metadataId.getNamespace(), metadataId.getPath() + ".json");
-		if (resourceManager.containsResource(metadataResourceId)) {
+		if (resourceManager.getResource(metadataResourceId).isPresent()) {
```

Comment:
This should be changed in a way to optimize IO interactions:

instead of inline, store the reference to the optional, keep the check and inside the try block reuse that optional resource to open it.
(also making it a try with resource would be nice since I definitely forgot that when I made the mode)